Xiaomi procedure failed: [getServiceToken] Missing serviceToken cookie

I followed the instructions to unlock my Xiaomi phone, but I found this procedure error in the current linux version and in the Windows version.

![Screenshot from 2021-07-13 09-35-19](https://user-images.githubusercontent.com/53882428/125452530-fa20405e-f2ac-4a04-9648-d3c2782ac931.png)

> `git clone https://github.com/francescotescari/XiaoMiToolV2`
> 
> **src/main/java/com/xiaomitool/v2/gui/controller/LoginController.java** Line 39:
> 
> Replace that line with:
> ` private static final String LOGIN_URL = "https://account.xiaomi.com/pass/serviceLogin?sid=unlockApi&json=false&passive=true&hidden=false&_snsDefault=facebook&checkSafePhone=true&_locale=" + Locale.getDefault().getLanguage().toLowerCase();`
> After that do `gradle build && gradle run` , login with 2FA verification, try to unlock your bootloader!
> 
> I've not tried myself because i don't want to use another smartphone for 2FA Verification and to Lock & ReUnlock the Bootloader, but if works, please mention me in this thread and tell me if you're able to unlock bootloader and mod your Xiaomi Smartphone!

I tried to do that:

```
$ git clone https://github.com/francescotescari/XiaoMiToolV2                                                               
Cloning into 'XiaoMiToolV2'...
remote: Enumerating objects: 2569, done.
remote: Counting objects: 100% (2566/2566), done.
remote: Compressing objects: 100% (778/778), done.
Receiving objects: 100% (2569/2569), 67.78 MiB | 388.00 KiB/s, done.
remote: Total 2569 (delta 1593), reused 2566 (delta 1593), pack-reused 3
Resolving deltas: 100% (1593/1593), done.
Updating files: 100% (332/332), done.
                                                                                                                                     
╭─[morfik@morfikownia] - [/media/debuilder/git-xiaomimitoolv2] - [2021-08-23 23:16:37]
╰─[0] < > $  cd XiaoMiToolV2
                                                                                                                                     
╭─[morfik@morfikownia] - [/media/debuilder/git-xiaomimitoolv2/XiaoMiToolV2] - [2021-08-23 23:20:27]
╰─[0] < > $  geany src/main/java/com/xiaomitool/v2/gui/controller/LoginController.java 

╭─[morfik@morfikownia] - [/media/debuilder/git-xiaomimitoolv2/XiaoMiToolV2] - [2021-08-23 23:20:47]
╰─[0] < > $  gradle build && gradle run
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :buildEnvironment

------------------------------------------------------------
Root project
------------------------------------------------------------

classpath
No dependencies


BUILD SUCCESSFUL in 12s
1 actionable task: 1 executed

FAILURE: Build failed with an exception.

* What went wrong:
Task 'run' not found in root project 'XiaoMiToolV2'.

* Try:
Run gradle tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
```

Should I do something else to make it work?

> > `git clone https://github.com/francescotescari/XiaoMiToolV2`
> > **src/main/java/com/xiaomitool/v2/gui/controller/LoginController.java** Line 39:
> > Replace that line with:
> > ` private static final String LOGIN_URL = "https://account.xiaomi.com/pass/serviceLogin?sid=unlockApi&json=false&passive=true&hidden=false&_snsDefault=facebook&checkSafePhone=true&_locale=" + Locale.getDefault().getLanguage().toLowerCase();`
> > After that do `gradle build && gradle run` , login with 2FA verification, try to unlock your bootloader!
> > I've not tried myself because i don't want to use another smartphone for 2FA Verification and to Lock & ReUnlock the Bootloader, but if works, please mention me in this thread and tell me if you're able to unlock bootloader and mod your Xiaomi Smartphone!
> 
> I tried to do that:
> 
> ```
> $ git clone https://github.com/francescotescari/XiaoMiToolV2                                                               
> Cloning into 'XiaoMiToolV2'...
> remote: Enumerating objects: 2569, done.
> remote: Counting objects: 100% (2566/2566), done.
> remote: Compressing objects: 100% (778/778), done.
> Receiving objects: 100% (2569/2569), 67.78 MiB | 388.00 KiB/s, done.
> remote: Total 2569 (delta 1593), reused 2566 (delta 1593), pack-reused 3
> Resolving deltas: 100% (1593/1593), done.
> Updating files: 100% (332/332), done.
>                                                                                                                                      
> ╭─[morfik@morfikownia] - [/media/debuilder/git-xiaomimitoolv2] - [2021-08-23 23:16:37]
> ╰─[0] < > $  cd XiaoMiToolV2
>                                                                                                                                      
> ╭─[morfik@morfikownia] - [/media/debuilder/git-xiaomimitoolv2/XiaoMiToolV2] - [2021-08-23 23:20:27]
> ╰─[0] < > $  geany src/main/java/com/xiaomitool/v2/gui/controller/LoginController.java 
> 
> ╭─[morfik@morfikownia] - [/media/debuilder/git-xiaomimitoolv2/XiaoMiToolV2] - [2021-08-23 23:20:47]
> ╰─[0] < > $  gradle build && gradle run
> Starting a Gradle Daemon (subsequent builds will be faster)
> 
> > Task :buildEnvironment
> 
> ------------------------------------------------------------
> Root project
> ------------------------------------------------------------
> 
> classpath
> No dependencies
> 
> 
> BUILD SUCCESSFUL in 12s
> 1 actionable task: 1 executed
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Task 'run' not found in root project 'XiaoMiToolV2'.
> 
> * Try:
> Run gradle tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> 
> * Get more help at https://help.gradle.org
> 
> BUILD FAILED in 1s
> ```
> 
> Should I do something else to make it work?

Wait few seconds, i'm reuploading it on DropBox.com!

Hello everybody!
I proceed all instructions(clone git repo, switch branch, change string) and now have error:
<code>
...
[10:39:51][INFO  ][4d725031] Starting unlock procedure
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f37070b3306, pid=29590, tid=29616
#
# JRE version: OpenJDK Runtime Environment (11.0.12+7) (build 11.0.12+7-post-Debian-2)
# Java VM: OpenJDK 64-Bit Server VM (11.0.12+7-post-Debian-2, mixed mode, sharing, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0x8d5306]
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /mnt/data/android/XiaoMiToolV2/hs_err_pid29590.log
#
# If you would like to submit a bug report, please visit:
#   https://bugs.debian.org/openjdk-11
#

> Task :run FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':run'.
> Process 'command '/usr/lib/jvm/java-11-openjdk-amd64/bin/java'' finished with non-zero exit value 134

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 4m 29s
4 actionable tasks: 2 executed, 2 up-to-date
</code>
What i missed or did wrong?

> Hello everybody!
> I proceed all instructions(clone git repo, switch branch, change string) and now have error:
> 
> ```
> ...
> [10:39:51][INFO  ][4d725031] Starting unlock procedure
> #
> # A fatal error has been detected by the Java Runtime Environment:
> #
> 
> #  SIGSEGV (0xb) at pc=0x00007f37070b3306, pid=29590, tid=29616
> #
> # JRE version: OpenJDK Runtime Environment (11.0.12+7) (build 11.0.12+7-post-Debian-2)
> # Java VM: OpenJDK 64-Bit Server VM (11.0.12+7-post-Debian-2, mixed mode, sharing, tiered, compressed oops, g1 gc, linux-amd64)
> # Problematic frame:
> # V  [libjvm.so+0x8d5306]
> #
> # No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
> #
> # An error report file with more information is saved as:
> # /mnt/data/android/XiaoMiToolV2/hs_err_pid29590.log
> #
> # If you would like to submit a bug report, please visit:
> #   https://bugs.debian.org/openjdk-11
> #
> 
> > Task :run FAILED
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Execution failed for task ':run'.
> > Process 'command '/usr/lib/jvm/java-11-openjdk-amd64/bin/java'' finished with non-zero exit value 134
> 
> * Try:
> Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> 
> * Get more help at https://help.gradle.org
> 
> BUILD FAILED in 4m 29s
> 4 actionable tasks: 2 executed, 2 up-to-date
> ```
> 
> What i missed or did wrong?
> Debian 11, Openjdk 11.0.12.

I had the same with my current Debian installation (sid/unstable). I don't know why it happens and how to fix it. You can download the latest ubuntu lts, and boot the live system and after you do all the steps again, it should work well.

> > Hello everybody!
> > I proceed all instructions(clone git repo, switch branch, change string) and now have error:
> > ```
> > ...
> > [10:39:51][INFO  ][4d725031] Starting unlock procedure
> > #
> > # A fatal error has been detected by the Java Runtime Environment:
> > #
> > 
> > #  SIGSEGV (0xb) at pc=0x00007f37070b3306, pid=29590, tid=29616
> > #
> > # JRE version: OpenJDK Runtime Environment (11.0.12+7) (build 11.0.12+7-post-Debian-2)
> > # Java VM: OpenJDK 64-Bit Server VM (11.0.12+7-post-Debian-2, mixed mode, sharing, tiered, compressed oops, g1 gc, linux-amd64)
> > # Problematic frame:
> > # V  [libjvm.so+0x8d5306]
> > #
> > # No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
> > #
> > # An error report file with more information is saved as:
> > # /mnt/data/android/XiaoMiToolV2/hs_err_pid29590.log
> > #
> > # If you would like to submit a bug report, please visit:
> > #   https://bugs.debian.org/openjdk-11
> > #
> > 
> > > Task :run FAILED
> > 
> > FAILURE: Build failed with an exception.
> > 
> > * What went wrong:
> > Execution failed for task ':run'.
> > > Process 'command '/usr/lib/jvm/java-11-openjdk-amd64/bin/java'' finished with non-zero exit value 134
> > 
> > * Try:
> > Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> > 
> > * Get more help at https://help.gradle.org
> > 
> > BUILD FAILED in 4m 29s
> > 4 actionable tasks: 2 executed, 2 up-to-date
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > What i missed or did wrong?
> > Debian 11, Openjdk 11.0.12.
> 
> I had the same with my current Debian installation (sid/unstable). I don't know why it happens and how to fix it. You can download the latest ubuntu lts, and boot the live system and after you do all the steps again, it should work well.

I test it on other pc(debian too), and have error 10000. Find many answers with possible solutions(log out, wait, log in). Wait now...

@andresangelini - thanks for summarizing well above :+1:  

I tried all of the steps but am getting stuck on step 9. When I run:
`./gradlew build && ./gradle run`

Everything seems to go well during the build phase. However when  `./gradle run` is executed I get the following error msg:

`> Task :compileJava FAILED
Caching disabled for task ':compileJava' because:
  Build cache is disabled

Task ':compileJava' is not up-to-date because:
  Input property 'classpath' file /root/.gradle/caches/modules-2/files-2.1/org.openjfx/javafx-fxml/11.0.2/1b8a331d5f393b48de0aef59b3c967a312967290/javafx-fxml-11.0.2-linux.jar has been removed.
  Input property 'classpath' file /root/.gradle/caches/modules-2/files-2.1/org.openjfx/javafx-web/11.0.2/49ca03e2fd58cbb0ad94f6a6bd11e0c043e541a7/javafx-web-11.0.2-linux.jar has been removed.
  Input property 'classpath' file /root/.gradle/caches/modules-2/files-2.1/org.openjfx/javafx-controls/11.0.2/5f6929050a744aab39ebfc1e8e5dd03bcd2ad47b/javafx-controls-11.0.2-linux.jar has been removed.
Full recompilation is required because unable to get source-classes mapping relationship from last compilation. Analysis took 0.0 secs.
:compileJava (Thread[Execution worker for ':',5,main]) completed. Took 0.139 secs.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
`

Got any input/feedback on what's going wrong here? Thank you for any help!
Like many others of you, I'm tired of being stuck behind the Xiaomi-walls =)

The jar file is the result of the compilation. To run it you need 

> > Hi guys ! Is it possible for one of you who succeded with the manipulation (modifiing the authentification method) to create the .jar file and to upload it somewhere ?
> > I'm trying to compile the solution but it doesn't work (i've never used gradle before)
> > Thanks in advance
> 
> Do you proceed README?
> Do you install propper JAVA?
> Show console output.


This is the cause of the errors : 

Caused by: java.lang.RuntimeException: java.lang.IllegalAccessError: class org.gradle.internal.compiler.java.ClassNameCollector (in unnamed module @0x605d5ad0) cannot access class com.sun.tools.javac.code.Symbol$TypeSymbol (in module j
dk.compiler) because module jdk.compiler does not export com.sun.tools.javac.code to unnamed module @0x605d5ad0

Caused by: java.lang.IllegalAccessError: class org.gradle.internal.compiler.java.ClassNameCollector (in unnamed module @0x605d5ad0) cannot access class com.sun.tools.javac.code.Symbol$TypeSymbol (in module jdk.compiler) because module
jdk.compiler does not export com.sun.tools.javac.code to unnamed module @0x605d5ad0


I only modified the authentification method and nothing else.

> > Sorry, if this is a silly question. I got this error, did I mess up something? Thanks
> > > Project : => no module-info.java found
> 
> Fixed by changing / setting default java to 11 version.

Hey man, i am having the same exact issue I get this:

`./gradlew build

> Configure project :
Project : => no module-info.java found

> Task :compileJava FAILED
Note: /home/aaron0791/Downloads/XiaoMi/XiaoMiToolV2/src/main/java/com/xiaomitool/v2/tasks/AdbPushTask.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> java.lang.IllegalAccessError: class org.gradle.internal.compiler.java.ClassNameCollector (in unnamed module @0x534003ba) cannot access class com.sun.tools.javac.code.Symbol$TypeSymbol (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.code to unnamed module @0x534003ba

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
1 actionable task: 1 executed
`

I have no idea what to do. You said you fixed your issue by changing / setting default java to 11 version, I am running java 16. Could you please tell me how to do this?

--- a/src/main/java/com/xiaomitool/v2/gui/controller/LoginController.java
+++ b/src/main/java/com/xiaomitool/v2/gui/controller/LoginController.java
@@ -36,7 +36,8 @@ import java.net.URI;
 import java.util.Locale;
 
 public class LoginController extends DefaultController {
-private static final String LOGIN_URL = "https://account.xiaomi.com/pass/serviceLogin?sid=unlockApi&json=false&passive=true&hidden=false&_snsDefault=facebook&checkSafePhone=true&_locale=" + Locale.getDefault().getLanguage().toLowerCase();    private static boolean loggedIn = false;
+private static final String LOGIN_URL = "https://account.xiaomi.com/pass/serviceLogin?sid=unlockApi&json=false&passive=true&hidden=false&_snsDefault=facebook&checkSafePhone=true&_locale=" + Locale.getDefault().getLanguage().toLowerCase();    
+    private static boolean loggedIn = false;
     private static Thread loginThread = null;
     @FXML
     private HBox HEADER;

--- a/src/main/java/com/xiaomitool/v2/gui/controller/LoginController.java
+++ b/src/main/java/com/xiaomitool/v2/gui/controller/LoginController.java
@@ -36,8 +36,7 @@ import java.net.URI;
 import java.util.Locale;
 
 public class LoginController extends DefaultController {
-    private static final String LOGIN_URL = "https://account.xiaomi.com/pass/serviceLogin?sid=passport&json=false&passive=true&hidden=false&_snsDefault=facebook&_locale=" + Locale.getDefault().getLanguage().toLowerCase();
-    private static boolean loggedIn = false;
+private static final String LOGIN_URL = "https://account.xiaomi.com/pass/serviceLogin?sid=unlockApi&json=false&passive=true&hidden=false&_snsDefault=facebook&checkSafePhone=true&_locale=" + Locale.getDefault().getLanguage().toLowerCase();    private static boolean loggedIn = false;
     private static Thread loginThread = null;
     @FXML
     private HBox HEADER;

